gateway/ln-gateway/src/db.rs

```diff
@@ -37,6 +41,26 @@ impl_db_record!(
 
 impl_db_lookup!(key = FederationIdKey, query_prefix = FederationIdKeyPrefix);
 
+impl FederationConfig {
+    pub fn to_gateway_registration_info(
+        &self,
+        route_hints: Vec<RouteHint>,
+        time_to_live: Duration,
+        node_pub_key: PublicKey,
+        api: Url,
+    ) -> LightningGateway {
+        LightningGateway {
+            mint_channel_id: self.mint_channel_id,
+            mint_pub_key: self.redeem_key.x_only_public_key().0,
+            node_pub_key,
+            api,
+            route_hints,
+            valid_until: fedimint_core::time::now() + time_to_live,
```

Comment:
The way we have an implicit `fedimint_core::time::now()` seems like a foot-gun to me. Let's leave it like this for now, but should fix this somehow ...

Reply:
Yes agreed

Reply:
https://github.com/fedimint/fedimint/issues/2651